Mathlib/Order/ConditionallyCompleteLattice/Basic.lean

```diff
@@ -1212,6 +1212,9 @@ theorem exists_lt_of_lt_ciSup' {f : ι → α} {a : α} (h : a < ⨆ i, f i) : 
   contrapose! h
   exact ciSup_le' h
 
+theorem not_mem_of_lt_csInf' {x : α} {s : Set α} (h : x < sInf s) : x ∉ s :=
```

Comment:
Can you add the `not_mem_of_csSup_lt`` version too?

Reply:
That wouldn't make much sense. The reason `not_mem_of_lt_csInf` and `not_mem_of_lt_csInf'` are distinct lemmas is because in the CC OrderBot case you can infer the `BddBelow` hypothesis; you can't do the same for `BddAbove`.

Reply:
I ended up needing this lemma in a separate PR, so I've PR'd it separately as #16988

Reply:
Isn't there symmetry with the CC OrderTop case?

Reply:
That typeclass doesn't exist, it seems like we haven't found any practical use for it.